Updated setup commands and Cross platform Setup (#120)
torchbox/wagtail-news-template: README.md

## Getting Started

### Check your Python version

**Check that you have an appropriate version of Python 3** You want to make sure that you have a [compatible version](https://docs.wagtail.org/en/stable/releases/upgrading.html#compatible-django-python-versions) installed:

```bash
python --version

# Or:
python3 --version

# On Windows:
py --version
```

### Navigate to the project directory:

```bash
cd myproject

> ```

Reviewer: We haven't instructed them to create this directory.

### Create and Activate virtual environment

#### Linux / macOS

```bash
python -m venv .venv
source .venv/bin/activate
```

#### Windows

```bash
python -m venv .venv
.venv\Scripts\activate
```

### Install Wagtail

Install the Wagtail CMS package using pip:

```bash
pip install wagtail
```

### Create a new project

Use the Wagtail News Template to generate a new project:

```bash
wagtail start --template=https://github.com/wagtail/news-template/archive/refs/heads/main.zip myproject .
```

### Bootstrap the project

Run the bootstrap command to install project dependencies and set up the development environment:

```bash
python bootstrap.py
```

This command will:

* Install project dependencies
* Run database migrations
* Create the cache table
* Load sample content
* Create the default administrator account
* Collect static files

> ### Optional: Skip sample content

Reviewer: Let's make this a subheading under "Bootstrap the project":

```suggestion
#### Optional: Skip sample content
```

If you prefer to start with a clean project without demo content:

```bash
python bootstrap.py --no-sample-data
```

This skips loading the sample content and default administrator account.

To create an administrator account manually, run:

```bash
python manage.py createsuperuser
```

### Start the development server

```bash
python manage.py runserver
```

### Access the site

Once the server is running:

* Site: `http://localhost:8000`

> * Admin: `http://localhost:8000/admin`

Reviewer: Just to avoid a redirect:

```suggestion
* Admin: `http://localhost:8000/admin/`
```

## Getting Started

1. **Check that you have an appropriate version of Python 3** You want to make sure that you have a [compatible version](https://docs.wagtail.org/en/stable/releases/upgrading.html#compatible-django-python-versions) installed:
### Check your Python version

   ```sh
   python --version
   # Or:
   python3 --version

>    # **On Windows** (cmd.exe, with the Python Launcher for Windows):

Reviewer: Is "cmd.exe, with the Python Launcher for Windows" important information? Could Python be invoked in other ways on Windows?

Reviewer 2: No it is not a necessary information .

python3 --version

# On Windows:
py --version
```

### Create a virtual environment

Create and activate a virtual environment for your project.

#### Linux / macOS

```bash
python -m venv .venv
source .venv/bin/activate
```

#### Windows

```bash

> python -m venv .venv

Reviewer: Above we're recommending that people use `py --version` on Windows. Does this need to be `py` here also?

.venv\Scripts\activate
```

### Install Wagtail

Install the Wagtail CMS package using pip:

```bash
pip install wagtail
```

### Create a new project

> Use the Wagtail News Template to generate a new project:

Reviewer: I think we should keep referring to it as the Starter Kit, since it seems like we're moving toward it not being specific to a news site.

```suggestion
Use the Wagtail Starter Kit to generate a new project:
```

> 

Reviewer: It might be nice to call out that `myproject` is just an example and they can name it whatever they want.

Reviewer 2: good idea

Make sure to test any changes by reviewing them against a newly created project, by following the [Getting Started](#getting-started) instructions again.

### Before Opening a Pull Request

Please install the project's pre-commit hooks:

```bash id="l8u5jy"
pre-commit install
```

Run the linting checks before submitting a pull request:

> ```bash id="vjlwmg"

Reviewer: What is the purpose of these `id`s?

Reviewer 2: My bad
